src/util/cmdlineargs.cpp

```diff
@@ -133,3 +136,23 @@ void CmdlineArgs::printUsage() {
 
     fputs("\n\n(For more information, see http://mixxx.org/wiki/doku.php/command_line_options)\n",stdout);
 }
+
+bool CmdlineArgs::getSettingsPathStatus() const {
+
+    QString path = getSettingsPath();
+    //info is neccessary as, only making of file doesn't tell about readablilty
+    QFileInfo info(path); 
+    bool status = true;
+    QTemporaryFile fle(path + "/temp_file");
+    if(fle.open()) {
+        fle.close();
+    }
+    else {
+        status = false;
+    }
+    if(!info.isReadable()) {
```

Comment:
Yes the file was created even without readable permission, and I thought is readable is a necessary condition so I had to add it extra as the making of file(writing) done above and the permission for execution(folder execution in linux) is checked by going to that directory and try to make the file, read permission tell whether we can execute comman ls in that directory or not, in simple sense, so I had to explicitly use the isReadable function